app/code/Magento/Backend/Model/Search/Config.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Backend\Model\Search;
+
+use Magento\Backend\Model\Search\Config\Result\Builder;
+use Magento\Config\Model\Config\Structure;
+use Magento\Config\Model\Config\Structure\Element\AbstractComposite;
+use Magento\Config\Model\Config\Structure\Element\Iterator as ElementIterator;
+
+/**
+ * Search Config Model
+ *
+ * @method Config setQuery(string $query)
+ * @method string|null getQuery()
+ * @method bool hasQuery()
+ * @method Config setResults(array $results)
+ * @method array getResults()
```

Comment:
it's better to have these methods in class contract, not relying on `__call` method